File: lib/models/weather.dart
import 'package:flutter_unity_challenge/utils/enum_convert.dart';

enum WeatherCondition{
  Thunderstorm,
  Drizzle,
  Rain,
  Snow,
  Atmosphere,
  Clear,
  Clouds
}

class Weather {
  int temperature;
  WeatherCondition weatherCondition;


  Weather({this.temperature, this.weatherCondition});

  factory Weather.fromJson(Map<String, dynamic> json) {
    return Weather(
      temperature: (json['main']['temp'] as num).round(),
      weatherCondition: enumFromString(json['weather'][0]['main'], WeatherCondition.values) ,
    );
  }

  Map<String, dynamic> toJson() =>
      {
        'temperature': temperature,
        'weatherCondition': weatherCondition != null ? enumToString(weatherCondition) : null,
      };
}

File: unity/FlutterUnityChallenge/Assets/Scripts/Entities/Weather.cs
using System;
using UnityEngine;

enum WeatherCondition {
    Thunderstorm,
    Drizzle,
    Rain,
    Snow,
    Atmosphere,
    Clear,
    Clouds
};

class Weather
{

    public int temperature;

    public string weatherCondition;

    public WeatherCondition WeatherConditionToEnum()
    {
        return (WeatherCondition)Enum.Parse(typeof(WeatherCondition), weatherCondition, true);
    }
    public string GetTemperatureWithUnit()
    {
        return string.Format("{0}º C", temperature);
    }

    public static Weather ParseJson(string json)
    {
        return JsonUtility.FromJson<Weather>(json);
    }
}